src/map/battle.c

```diff
@@ -6627,6 +6627,39 @@ static bool battle_check_arrows(struct map_session_data *sd)
 	return true;
 }
 
+/**
+ * Check whether an attack from src to target should cause Blade Stop to start.
+ * @param src attacker
+ * @param target unit being attacked
+ * @returns true if Blade Stop should start, false otherwise
+ */
+static bool battle_should_bladestop_attacker(struct block_list *src, struct block_list *target)
+{
+	nullpo_retr(false, src);
+	nullpo_retr(false, target);
+
+	struct status_change *tsc = status->get_sc(target);
+	if (tsc == NULL || tsc->data[SC_BLADESTOP_WAIT] == NULL)
+		return false; // Target is not in BladeStop wait mode
+
+	if (is_boss(src))
+		return false; // Boss monsters are not affected
+
+	// CHECKME: Is that right?
+	if (src->type == BL_PC)
+		return true; // Player gets into BladeStop regardless of distance
```

Comment:
I tested this for RE and this behavior is wrong.

In Aegis, if a ranged monster (e.g. Orc Archer) or a ranged player (e.g. a Hunter) hits me from many cells away, Blade Stop won't start.

I will hold the fix for after rebalance to avoid adding non-rebalance things into the mix of this PR chain.

@skyleo may you check how this works for pre-re?

Reply:
copy from our chat:
```
So yeah, if attacker player and attacking bladestopped player -> No range check. if attacking mob -> no bladestop check

If attacker mob or anything else non-player and attacking bladestopped anything -> range check of 2 

```

Reply:
In Aegis 11.2 the code is written in a way that it assumes that mobs cannot use Bladestop, so when someone is in Bladestop there is code that assumes that someone is a player, without checking if it's player type.

Reply:
Thank you very much! moved the refactor + fix to #3254 so it may go to master earlier. And I will rebase this PR later on